File: apps/backend/src/services/reservations/confirmationCode.ts
export function makeConfirmationCode(prefix: string = 'DEMO'): string {
  // Simple deterministic-ish code: DEMO-XXXXXX
  const chars = 'ABCDEFGHJKLMNPQRSTUVWXYZ23456789';
  let s = '';
  for (let i = 0; i < 6; i++) {
    s += chars[Math.floor(Math.random() * chars.length)];
  }
  return `${prefix}-${s}`;
}

File: python/backend/app/services/confirmation_code.py
from __future__ import annotations

import random


def make_confirmation_code(prefix: str = "DEMO") -> str:
    chars = "ABCDEFGHJKLMNPQRSTUVWXYZ23456789"
    return f"{prefix}-" + "".join(random.choice(chars) for _ in range(6))
